packages/core/src/hooks/data/useCustom.ts

```diff
@@ -58,18 +61,19 @@ export const useCustom = <
     successNotification,
     errorNotification,
     metaData,
+    dataProviderName,
 }: UseCustomProps<TData, TError, TQuery, TPayload>): QueryObserverResult<
     CustomResponse<TData>,
     TError
 > => {
-    const { custom } = useContext<IDataContext>(DataContext);
+    const { custom } = useDataProvider(dataProviderName);
     const { mutate: checkError } = useCheckError();
     const translate = useTranslate();
     const handleNotification = useHandleNotification();
 
     const queryResponse = useQuery<CustomResponse<TData>, TError>(
         [`custom/${method}-${url}`, { ...config, ...metaData }],
-        () => custom<TData>({ url, method, ...config, metaData }),
+        () => custom!<TData>({ url, method, ...config, metaData }),
```

Comment:
```suggestion
        () => custom?<TData>({ url, method, ...config, metaData }),
```
"custom" function can be undefined